### DISCUSSION TITLE
Reboot project

### DISCUSSION THREAD

**Discussion by drieseng** [sshnet/SSH.NET#1044] [General]
Let's us this as a starting point for discussing how we can reboot/revive this project.

Before you consider joining this project, I'd like to make it clear that it will be hard to convince me to accept features or bug fixes that are not covered by unit tests and/or integration tests. New features should also be generally useful; not just for yourself.

If you read past the previous section and you're still interested: hurray! :-)

I'm very much aware that I risk becoming a bottleneck, but I'll try my very best to keep up with you.
If - after some time - this is identified as being an issue, we'll discuss how to proceed.
Early on, I'll surely review PR's in more detail and I expect the same from all of you.

@ZybexXL @IgorMilavec @daviburg

**Comment by daviburg** [comment]
Absolutely. Test coverage is a must. Features should be justifiably beneficial albeit that might be a bit subjective.

With regards,

David.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Gert Driesen ***@***.***>
Sent: Saturday, November 12, 2022 6:32:24 AM
To: sshnet/SSH.NET ***@***.***>
Cc: David Burg ***@***.***>; Mention ***@***.***>
Subject: [sshnet/SSH.NET] Reboot project (Discussion #1044)


Let's us this as a starting point for discussing how we can reboot/revive this project.

Before you consider joining this project, I'd like to make it clear that it will be hard to convince me to accept features or bug fixes that are not covered by unit tests and/or integration tests. New features should also be generally useful; not just for yourself.

If you read past the previous section and you're still interested: hurray! :-)

I'm very much aware that I risk becoming a bottleneck, but I'll try my very best to keep up with you.
If - after some time - this is identified as being an issue, we'll discuss how to proceed.
Early on, I'll surely review PR's in more detail and I expect the same from all of you.

@zybexXL<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FzybexXL&data=05%7C01%7Cdaviburg%40microsoft.com%7C5b7ef39b44d14d5a31fb08dac4bab7c0%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638038603470065828%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=aifqNkD6t5rtOPDdkHnMwCsHcH5h7IQJFA8r4FNwZAA%3D&reserved=0> @IgorMilavec<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FIgorMilavec&data=05%7C01%7Cdaviburg%40microsoft.com%7C5b7ef39b44d14d5a31fb08dac4bab7c0%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638038603470065828%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=utmNqaiJrRRspXD7ny0dDMoyYyHPg4Qv1sntvG0jU2o%3D&reserved=0> @daviburg<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdaviburg&data=05%7C01%7Cdaviburg%40microsoft.com%7C5b7ef39b44d14d5a31fb08dac4bab7c0%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638038603470065828%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=AWoEZ5CzGtnAjFp2qoLAEZ2pGNUZaJIHgMPGChwJcag%3D&reserved=0>

**Comment by daviburg** [comment]
Are there specific PRs in waiting we could help with?

**Comment by zybexXL** [comment]
Agreed with the unit tests and general feature usefulness. 

However, how do you see the workflow improving when you will still be the single one doing final approvals? If other moderators are just here to do reviews and engage with contributors, while you alone retain the merge permission... then nothing changes.

**Reply by drieseng** [reply]
I propose to start like this. Anyone can of course do reviews.
I expect / hope our views will start to align after some time.
When enough trust has been built, I will of course allow others to approve merge requests.

**Comment by zybexXL** [comment]
That is reasonable.

**Reply by drieseng** [reply]
@zybexXL, @daviburg  PR #910 is a small one to start with. The last CI build for that PR failed, but it may be a flaky unit test.

I've identified the following tasks for this PR:

1. Review the changes.
1. Attempt to reproduce the test failure and/or investigate why the test is flaky.
1. Add integration test for the mentioned scenario (negative seek at EOF).
1. Document the fix in draft release notes for the next release.

For the third task, I've granted you both access to the **IntegrationTests** repo.
That repo is not public, but I'm ok with opening it up if you agree that it's useful and offers a high enough level of quality.
In a next phase, we could consider integrating it into the main repo.
